### PR TITLE
Make it possible to configure CronetEngine settings

### PR DESCRIPTION
**Pull request by brianquinlan**

### CODE REVIEW

File: pkgs/cronet_http/lib/cronet_client.dart [hunk 1]
```diff
@@ -34,10 +122,27 @@ import 'src/messages.dart';
...
+  CronetEngine? _engine;
+  final bool _ownedEngine;
+
+  CronetClient([CronetEngine? engine])
+      : _engine = engine,
+        _ownedEngine = engine == null;
+
+  @override
+  void close() {
+    if (_ownedEngine) {
```
Comment (aam) [inline pkgs/cronet_http/lib/cronet_client.dart:134]: would `_engine != null` be sufficient here making _ownedEngine property unnecessary?

File: pkgs/cronet_http/lib/src/messages.dart [hunk 2]
```diff
@@ -219,6 +332,60 @@ class HttpApi {
...
+        code: (error['code'] as String?)!,
+        message: error['message'] as String?,
+        details: error['details'],
+      );
+    } else if (replyMap['result'] == null) {
+      throw PlatformException(
+        code: 'null-error',
+        message: 'Host platform returned null value for non-null return value.',
+      );
+    } else {
```
Comment (aam) [inline pkgs/cronet_http/lib/src/messages.dart:360]: You could drop `else` block reducing nesting/indentation.

Comment (aam) [reply pkgs/cronet_http/lib/src/messages.dart:360]: here and below.

File: pkgs/cronet_http/example/integration_test/cronet_configuration_test.dart [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+          request.response.headers.set('etag', '12345');
+          await request.response.close();
+        });
+    });
+    tearDown(() {
+      server.close();
+    });
+
+    test('disabled', () async {
+      final builder = await CronetEngine.build(cacheMode: CacheMode.disabled);
```
Comment (aam) [inline pkgs/cronet_http/example/integration_test/cronet_configuration_test.dart:36]: I think these are engines, not builders (here and below).

**Review by aam — APPROVED** [review]

File: pkgs/cronet_http/lib/cronet_client.dart (hunk 1 above)
Comment (brianquinlan) [reply pkgs/cronet_http/lib/cronet_client.dart:134]: `_engine` gets assigned in `send` if not set in the constructor.

File: pkgs/cronet_http/lib/src/messages.dart (hunk 2 above)
Comment (brianquinlan) [reply pkgs/cronet_http/lib/src/messages.dart:360]: This is generated code.

File: pkgs/cronet_http/example/integration_test/cronet_configuration_test.dart (hunk 3 above)
Comment (brianquinlan) [reply pkgs/cronet_http/example/integration_test/cronet_configuration_test.dart:36]: Done

**Pull request merged**